new_reduce.rb

```diff
@@ -0,0 +1,9 @@
+class Array
+  def my_reduce(joint)
+    if block_given?
+      self.each do |i|
+        # something clever
```

Comment:
I did some more reading and futzing but didn't make headway.  I'm working on the CF code challenge but will spend some more time on it later tonight.